common/src/main/java/io/netty/util/concurrent/SaturationHistogram.java

```diff
@@ -0,0 +1,70 @@
+/*
+ * Copyright 2012 The Netty Project
+ *
+ * The Netty Project licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package io.netty.util.concurrent;
+
+import java.util.concurrent.atomic.AtomicInteger;
+import java.util.concurrent.atomic.AtomicLong;
+
+public class SaturationHistogram {
```

Comment:
why not using https://github.com/HdrHistogram/HdrHistogram?
It's very cheap in space occupation and very fast too

Reply:
I had considered it, but didn't want to add a new dependency just for it.  The implementation here doesn't need to be as complicated as the HdrHistogram since its tracking a known number of contiguous buckets.